core/types/tx_blob.go

```diff
@@ -326,98 +280,72 @@ func (tx *BlobTx) withSidecar(sideCar *BlobTxSidecar) *BlobTx {
 }
 
 func (tx *BlobTx) encode(b *bytes.Buffer) error {
-	switch {
-	case tx.Sidecar == nil:
-		return rlp.Encode(b, tx)
-
-	case tx.Sidecar.Version == BlobSidecarVersion0:
-		return rlp.Encode(b, &blobTxWithBlobsV0{
-			BlobTx:      tx,
-			Blobs:       tx.Sidecar.Blobs,
-			Commitments: tx.Sidecar.Commitments,
-			Proofs:      tx.Sidecar.Proofs,
-		})
+	err := rlp.Encode(b, tx)
+	if err != nil {
+		return err
+	}
 
-	case tx.Sidecar.Version == BlobSidecarVersion1:
-		return rlp.Encode(b, &blobTxWithBlobsV1{
-			BlobTx:      tx,
-			Version:     tx.Sidecar.Version,
-			Blobs:       tx.Sidecar.Blobs,
-			Commitments: tx.Sidecar.Commitments,
-			Proofs:      tx.Sidecar.Proofs,
-		})
+	type sidecarNoVersion struct {
+		Blobs       []kzg4844.Blob       // Blobs needed by the blob pool
+		Commitments []kzg4844.Commitment // Commitments needed by the blob pool
+		Proofs      []kzg4844.Proof      // Proofs needed by the blob pool
+	}
 
-	default:
-		return errors.New("unsupported sidecar version")
+	if sidecar := tx.Sidecar; sidecar != nil {
+		if sidecar.Version != BlobSidecarVersion0 {
```

Comment:
It breaks the format we use now, making it incompatible with the existing blob data.